sockeye/inference.py

```diff
@@ -395,9 +395,8 @@ def load_models(context: mx.context.Context,
                                    restrict lexicon).
     :param forced_max_output_len: An optional overwrite of the maximum output length.
     :param override_dtype: Overrides dtype of encoder and decoder defined at training time to a different one.
-    :param output_scores: Whether the scores will be needed as outputs. If True, scores will be normalized, negative
-           log probabilities. If False, scores will be negative, raw logit activations if decoding with beam size 1
-           and a single model.
+    :param skip_softmax: If False, scores will be returned by the decoder step as normalized negative log probabilities.
```

Comment:
No, I had to update it. Softmax is now skipped if scores are requested or if sampling is turned on. Since there were two variables, it was more efficient to make the decisions outside the loop and then pass that in.

Reply:
Ah got it, thanks! However, if we think of `load_models()` as an API/library call it may be better to move the logic into there instead of relying on cmdline args. Sampling is kind of similar to the existing variables of this function (skip output layer for vocab restriction etc.) Not sure though, so your call.
In any case, I liked the name `output_scores` (and its docstring mentioning the conditions when this can be active) more than `skip_softmax`, which is easy to confuse with `skip_topk`.